clash-lib/src/Clash/Core/Evaluator.hs

```diff
@@ -426,6 +426,32 @@ instantiate h k (TyLambda x e) ty = (h,k,substTm "Evaluator.instantiate" subst e
   is0    = mkInScopeSet (localFVsOfTerms [e] `unionUniqSet` tyFVsOfTypes [ty])
 instantiate _ _ _ _ = error "not a ty lambda"
 
+naturalLiteral :: Value -> Maybe Integer
```

Comment:
Could you use these functions in `Clash.GHC.Evaluator`?

Reply:
Yes, done.